### DISCUSSION TITLE
Validation: How to customize localization messages in Password rules?

### DISCUSSION THREAD

**Discussion by pronist** [laravel/framework#41915] [Q&A] [answered]
I am trying to customize localization messages in Password rules. but, I think password validation error messages only support English. How to change error messages to other languages without  `passes()` method override?

https://github.com/laravel/framework/blob/78cb588f9f391016632b3f96755c851a712a442f/src/Illuminate/Validation/Rules/Password.php#L305-L319

**Comment by Nielson** [comment, marked as answer]
Hi @pronist 

I was just struggling with this as well. I found the answer in this post: https://www.magutti.com/blog/laravel-8-password-rule.

You have to create a .json file like "en.json" or whatever your localization is, in your lang folder. 

Eg:

```
/resources
    /lang
        it.json
        es.json
        en.json
```

This must contain the following:

```json
{
    "The :attribute must contain at least one uppercase and one lowercase letter.": "Your translation",
    "The :attribute must contain at least one letter.": "Your translation",
    "The :attribute must contain at least one symbol.": "Your translation"
}
```

I hope this helps :)

**Reply by mariankustusch** [reply]
In laravel 9 /lang is no longer in /resources folder, but rather in main folder.

**Comment by pronist** [comment]
@Nielson Excellent! It works for me.

**Reply by Kamandlou** [reply]
Maybe this will help you https://github.com/Laravel-Lang/lang
